### DISCUSSION TITLE
How do I setup PROVISIONING_PROFILE_SPECIFIER for both: host app and share extension?

### DISCUSSION THREAD

**Discussion by kunass2** [fastlane/fastlane#19807] [Q&A]
In my ios app I have used so far only one single app. And everything was working correctly. But since I have added share extension, things got complicated a little;) 

I have function for archiving defined like this:

```
def archive(options)
  prepare_for_archive(options)
  build_ios_app(
    workspace: PLZ_WORKSPACE,
    scheme: options[:scheme],
    clean: true,
    export_method: options[:adhoc] ? "ad-hoc" : "app-store",
    output_directory: OUTPUT_PATH,
    export_options: {
      provisioningProfiles: { 
        options[:bundle_id] => options[:provisioning],
        options[:share_bundle_id] => options[:share_provisioning]
      },
    },
    xcargs: { 
      :PROVISIONING_PROFILE_SPECIFIER => options[:provisioning],
    },
  )
end
```

But I got an error, because PROVISIONING_PROFILE_SPECIFIER do not indicate provisioning for my share extension. How can I do it here? I am not strong enough with ruby, but suppose you can help me with this pretty easy. Thank you.

[Here](https://stackoverflow.com/questions/70739761/how-do-i-sign-share-extension-along-with-the-host-app-using-fastlane)  is related stack overflow question for my problem:

**Comment by lucgrabowski** [comment]
I think that you should update code signing in xcode project for both targets - main app and share extension before building_ios_app. Note that when you call `update_project_provisioning` twice http://docs.fastlane.tools/actions/update_project_provisioning/#update_project_provisioning, you should use `target_filter` (regexp or string), e.g. "^" + <TARGET_NAME> + "$". Otherwise you will change settings twice for all targets and both main app and extension will have the same settings and the build will fail.

**Comment by Keon241** [comment]
The locations of any sparse SDKs that should be layered on top of the one specified by [Base SDK (SDKROOT)](https://xcodebuildsettings.com/itcaec37c2a6.html#devabcaadbaf). If more than one SDK is listed, the first one has highest precedence. Every SDK specified in this setting should be a “sparse” SDK, for example, not an SDK for an entire macOS release

[scentsy workstation](https://www.scentsyworkstation.net/)

**Comment by kntkymt** [comment]
Hi 👋

I encountered the simullar issue and I found solution.

- add `skip_profile_detection: true` to `build_ios_app`
- remove `xcargs: {  :PROVISIONING_PROFILE_SPECIFIER => options[:provisioning], },`

for my call
```ruby
gym(
      configuration: "Release",
      derived_data_path: "./DerivedData",
      skip_profile_detection: true,
      clean: true,
      verbose: true,
      scheme: "Release",
      export_method: "app-store",
      output_directory: "./build/ipa/" + Time.new.strftime("%Y/%m/%d/%H%M"),
      output_name: "Release.ipa",
      include_bitcode: true,
      xcargs: "OTHER_SWIFT_FLAGS='$(inherited) -DSTGING' CODE_SIGN_IDENTITY='" + ENV["CODE_SIGN_IDENTITY"] + "' FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD='" + ENV["FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD"] + "' FASTLANE_SESSION='" + ENV["FASTLANE_SESSION"] + "'",
      export_xcargs: "-allowProvisioningUpdates",
      export_options: {
        method: "app-store",
        compileBitcode: true,
        uploadBitcode: true,
        provisioningProfiles: {
          "sample.app" => ENV["PROVISIONING_PROFILE_SPECIFIER"],
          "sample.app.widget" => ENV["WIDGET_PROVISIONING_PROFILE_SPECIFIER"]
        }
      }
    )
```

(I don't know your issue and my issue are the same)
my problem is fastlane override `provisioningProfiles` mapping by `profile_detection`. It can be disable by `skip_profile_detection: true` to `build_ios_app`
